xml/obs_ag_security_concepts.xml

```diff
@@ -181,4 +181,140 @@
+   </sect3>
+   <sect3 xml:id="untrusted_code">
+    <title>Untrusted Code</title>
+    <para>
+      All code which is used to build content is considered
+      to be untrusted code. This includes even the Linux kernel, since
+      users can build and run their own kernels. The security layer here
+      is usually either KVM and XEN virtualization. These instances are
+      running without any network enabled. Build results get copied by 
+      reading a simple blocklist directly from the device. A security 
+      inside of the file system code can therefore not be used to 
+      compromise the worker.
+    </para>
+   </sect3>
+  </sect2>
+  <sect2 xml:id="dmz">
+   <title>Demilitarized Zone (DMZ)</title>
+   <para>The Demilitarized Zone contains service which interacts with
+         the public zone directly. 
+   </para>
+   <sect3 xml:id="frontend">
+    <title>&obs; Frontend</title>
+    <para>
+      The frontend service is the only service which provides an open port.
+      It implements the authorization of user requests. The authentification
+      may happen in an external service like a proxy or ldap server.
+      The frontend reads and writes changes to the database and
+      the source server only.
+    </para>
+   </sect3>
+   <sect3 xml:id="frontend_background">
+    <title>&obs; Frontend Background</title>
+    <para>
+     &obs; frontend background services handle less time critical operations.
+     This includes services which read data from external services like bug
+     trackers, sending notifications or long running jobs.
+    </para>
+   </sect3>
+   <sect3 xml:id="stage_server">
+    <title>Stage Server</title>
+    <para>
+      The stage server is providing the public content of the &obs; backends.
+      The server can be public accessable or just an upload server to
+      a mirror infrastructure.
+    </para>
+   </sect3>
+   <sect3 xml:id="cloud_uploader">
+    <title>Cloud Uploader</title>
+    <para>
+      The cloud uploader handles build results similar to the publisher.
+      This is an optional service.
+    </para>
+   </sect3>
+   <sect3 xml:id="source_service">
+    <title>Source Service Server</title>
+    <para>
+      The source service server is acting based on uploaded sources. The 
+      services should be written with security in mind and the administrator
+      can decide which services are trustable. The &obs; shipment defines
+      a minimum set of especially trusted services which received a security review.
+      A container might be used for additional security for each service run.
+    </para>
+   </sect3>
+  </sect2>
+  <sect2 xml:id="internal_zone">
+   <title>Internal Zone</title>
+   <para>The internal zone is running service which are supposed to work
+         without further external dependency.
+   </para>
+   <sect3 xml:id="source_server">
+    <title>&obs; Source Server</title>
+    <para>
+      A single source server is synchronizing all actions from the DMZ.
```

Comment:
while there is the information where the source server is syncing from, i am missing the information where the source server is syncing _to_.

Reply:
it is not syncing content at all, it just ensures that events get distributed to the right places.

Reply:
I completely misunderstood the sentence initially--I thought the source server was syncing something from the DMZ to somewhere else.

But if I am now understanding it correctly, the source server is sitting in the DMZ and acting to make sure events/actions are executed in a coordinated fashion (aka "in sync").

I guess in light of that misunderstanding, it would probably make sense to rephrase the sentence. Maybe pull the DMZ to the beginning of the sentence to make it clearer, like this:
`From within the DMZ, the source server coordinates changes to package and project configuration. There can only be a single source server per OBS install.` 
(I am not making this an official suggestion, since I am not 100% sure of the wording yet.)

Reply:
please check again, I made another attempt based on your suggestion.

Reply:
The new version works for me in terms of wording--though the sentence `Any source change is tracked and hosts the revision history.` seems like a word is missing somewhere before "hosts".

However, this new version now omits any mention of the DMZ--it is categorized into the DMZ section of the document but I figure, that the guide should probably attempt to explain why certain things need to be in a specific zone, rather than explaining the functionality at length, which is better done in a section like the component overview from the Reference Guide. But maybe that is something for another day.

I also finally looked again at my branch from 2017 that was going to move the component overview section out of the Reference Guide and into Admin Guide. I am going to try to revive that (right now, it's too bitrotten to rebase on master in any reasonable way though).

Reply:
changed it again.

the source server is part of the "internal network", not the DMZ where the api lives.